lib/rust/parser/src/macros/pattern.rs

```diff
@@ -341,6 +357,23 @@ impl Pattern {
                         Err(input)
                     },
             },
+            PatternData::Block => match input.pop_front() {
```

Comment:
Looks like we have some _parser combinators_, here! @hubertp is currrently trying to get rid of one such library as part of #9089...